### ISSUE TITLE
Dynamic Translations (e.g. via network)

### ISSUE THREAD

**Issue by Tienisto** [Tienisto/slang#52]
**Motivation**
To reduce asset size, developers can extract the translations to the server.
The translations will be fetched at **runtime**.

**Developer Experience**
Because every key needs to be known at compile time, the base translations must be available.
```json5
// strings.i18n.json
{
  "greet": "Welcome {name}"
}
```

Now load a locale:
```dart
// your logic, represents ONE locale!
Map fetchFromServer() {
  return {
    'greet': 'Hello {name}'
  }
}

// via LocaleSettings
void example1() {
  String a = t.greet(name: 'Tom'); // "Welcome Tom"

  Map data = fetchFromServer(); 
  LocaleSettings.loadCustom(data); // the new load function
  String b = t.greet(name: 'Tom'); // "Hello Tom"
}

// via dependency injection
void example2() {
  String a = t.greet(name: 'Tom'); // "Welcome Tom"

  Map data = fetchFromServer();
  final t2 = AppLocaleUtils.buildCustom(data); // the new build function
  String b = t2.greet(name: 'Tom'); // "Hello Tom"
}
```

This will blow up the generated file. Therefore, this should be opt-in.
```yaml
targets:
  $default:
    builders:
      fast_i18n:
        options:
          contexts:
            dynamic_translations: true # enable this feature
```

**Comment by erabti** [comment]
When can we expect this to land @Tienisto

**Comment by erabti** [comment]
Yes, reducing asset size and the ability to change strings remotely.

**Comment by Tienisto** [comment]
Released in `v3.0.0-dev.0`.
